pyproject.toml

```diff
@@ -3,19 +3,19 @@ name = "rockcraft"
 description = "Create rocks"
 dynamic = ["version"]
 dependencies = [
-    "craft-application~=5.6",
-    "craft-archives~=2.1",
-    "craft-cli~=3.1",
-    "craft-parts~=2.20",
-    "craft-platforms~=0.10",
+    "craft-application>=5.7.1",
+    "craft-archives>=2.2.0",
+    "craft-cli>=3.1.2",
+    "craft-parts>=2.20.1",
+    "craft-platforms>=0.10.0",
     # craft-providers is pinned to 2.2 until this is fixed:
     # https://github.com/canonical/craft-providers/issues/753
-    "craft-providers~=2.4",
-    "overrides",
+    "craft-providers>=3.0.0",
```

Comment:
I moved the version specifiers from `~=` to  `>=` because originally I completely missed the craft-providers release. Since ultimately we're using the lockfile I feel like specifying minimum versions might be a better approach, as typically when updating the lockfile we probably want to at least be made aware of the latest versions of the dependencies.